### PR TITLE
cmd/snap-update-ns: manually implement isspace

### PR DESCRIPTION
**Pull request by zyga**
I wasn't able to find any other cause but using the isspace macro,
outside of tests (which pass just fine), causes ./snap-update-ns -u 1000
to crash. The debugger points to issspace(*uid_text) being the culprit
despite it not being a NULL pointer. It is correctly pointing to the
string "1000" when the crash happens. I was unable to solve it with a
static function that calls isspace internally (which is probably a
macro). As such, implement isspace ourselves to avoid the crash.

Upstream bug report: https://github.com/golang/go/issues/29689
also contains a minimal reproducer https://github.com/zyga/go-bug-29689
which crashes on latest golang binary release.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>

### CODE REVIEW

**Comment by zyga** [comment]
We have a reply on the bug report. This is the way things will misbehave if we use pre-init array. This is not a Go bug.

File: cmd/snap-update-ns/bootstrap.c [hunk 1]
```diff
@@ -340,11 +340,14 @@ static int parse_arg_u(int argc, char * const *argv, int *optind, unsigned long
...
 	char *uid_text_end = NULL;
 	unsigned long parsed_uid = strtoul(uid_text, &uid_text_end, 10);
+	char c = *uid_text;
 	if (
 			/* Reject overflow in parsed representation */
 			(parsed_uid == ULONG_MAX && errno != 0)
 			/* Reject leading whitespace allowed by strtoul. */
-			|| (isspace(*uid_text))
+			/* NOTE: This is not using isspace, for some reason that is crashing
+			 * at runtime. Reported upstream as https://github.com/golang/go/issues/29689 */
```
Comment (bboozzoo) [inline cmd/snap-update-ns/bootstrap.c]: Comment needs an update now.

Comment (zyga) [reply cmd/snap-update-ns/bootstrap.c]: Done

**Comment by zyga** [comment]
I will implement islower/isspace now.

**Pull request merged**